main.go

```diff
@@ -0,0 +1,543 @@
+	//       <TimeNextVisible>...</TimeNextVisible>
+	//       <DequeueCount>...</DequeueCount>
+	//       <MessageText>...</MessageText>
+	//     </QueueMessage>
+	//     ...
+	// </QueueMessagesList>
+	// ```
+	// We unmarshal the response into go objects, using the go xml decoder.
+	fullBody, err := ioutil.ReadAll(resp.Body)
+	if err != nil {
+		return nil, err
+	}
+	reader := strings.NewReader(string(fullBody))
+	dec := xml.NewDecoder(reader)
+	err = dec.Decode(&queueMessagesList)
+	if err != nil {
+		return nil, err
+	}
+	if len(queueMessagesList.QueueMessages) == 0 {
+		return nil, errors.New("Zero tasks returned in Azure XML QueueMessagesList")
+	}
+	if size := len(queueMessagesList.QueueMessages); size > 1 {
+		return nil, fmt.Errorf("%v tasks returned in Azure XML QueueMessagesList, even though &numofmessages=1 was specified in poll url", size)
+	}
+
+	// at this point we know there is precisely one QueueMessage (== task)
+	qm := queueMessagesList.QueueMessages[0]
+
+	// To find the task referenced in a message the worker must base64
+	// decode and JSON parse the contents of the <MessageText> tag. This
+	// would return an object on the form: {taskId, runId}.
+	m, err := base64.StdEncoding.DecodeString(qm.MessageText)
+	if err != nil {
+		return nil, err
+	}
+
+	// Utility method for replacing a placeholder within a uri with
+	// a string value which first must be uri encoded...
+	detokeniseUri := func(uri, placeholder, rawValue string) string {
+		return strings.Replace(uri, placeholder, strings.Replace(url.QueryEscape(rawValue), "+", "%20", -1), -1)
+	}
+
+	// Before using the signedDeleteUrl the worker must replace the placeholder
+	// {{messageId}} with the contents of the <MessageId> tag. It is also
+	// necessary to replace the placeholder {{popReceipt}} with the URI encoded
+	// contents of the <PopReceipt> tag.  Notice, that the worker must URI
+	// encode the contents of <PopReceipt> before substituting into the
+	// signedDeleteUrl. Otherwise, the worker will experience intermittent
+	// failures.
+
+	// Since urlPair is a value, not a pointer, we can update this copy which
+	// is associated only with this particular task
+	urlPair.SignedDeleteUrl = detokeniseUri(
+		detokeniseUri(
+			urlPair.SignedDeleteUrl,
+			"{{messageId}}",
+			qm.MessageId,
+		),
+		"{{popReceipt}}",
+		qm.PopReceipt,
+	)
+
+	// initialise fields of TaskRun not contained in json string m
+	taskRun := TaskRun{
+		QueueMessage:  qm,
+		SignedURLPair: urlPair,
+	}
+	// now populate remaining json fields of TaskRun from json string m
+	err = json.Unmarshal(m, &taskRun)
+	if err != nil {
+		return nil, err
+	}
+
+	// Workers should read the value of the `<DequeueCount>` and log messages
+	// that alert the operator if a message has been dequeued a significant
+	// number of times, for example more than 10.
+	if qm.DequeueCount > 10 {
+		log.Printf("WARN: Task with id %v has been dequeued %v times!", taskRun.TaskId, qm.DequeueCount)
```

Comment:
Do this before you decode the JSON message.
In fact do it as soon as you have constructed the `signedDeleteUrl`.
If the message has been dequeued `15+` times, you should log and delete the message, remember to delete with retries...